### PR TITLE
Adds documentation for OIDC provider specific extension for G Suite

### PR DESCRIPTION
**Pull request by austingebauer**
This PR adds documentation for using an OIDC provider specific extension for G Suite that allows for fetching of groups and user information during authentication. Please see the rendered [documentation preview](https://5f08d685c1a9d20007bb58da--vault-www.netlify.app/docs/auth/jwt_oidc_providers#google-specific-handling-configuration).

Associated PR for the feature: https://github.com/hashicorp/vault-plugin-auth-jwt/pull/123

### CODE REVIEW

File: website/pages/docs/auth/jwt_oidc_providers.mdx [hunk 1]
```diff
@@ -128,6 +128,79 @@ Main reference: [Using OAuth 2.0 to Access Google APIs](https://developers.googl
 1. Configure Authorized Redirect URIs.
 1. Save client ID and secret.
 
+### Google-specific handling configuration
+
+Provider specific configuration is available when using Google as an identity provider from the
+Vault JWT/OIDC auth method. The configuration allows Vault to obtain G Suite group membership and
+user information during the JWT/OIDC authentication flow. The group membership obtained from G Suite
+will be used for Identity group alias association. The user information obtained from G Suite can be
```
Comment (kalafut) [inline website/pages/docs/auth/jwt_oidc_providers.mdx:136]:
```suggestion
may be used for Identity group alias association. The user information obtained from G Suite can be
```

File: website/pages/docs/auth/jwt_oidc_providers.mdx [hunk 2]
```diff
@@ -128,6 +128,79 @@ Main reference: [Using OAuth 2.0 to Access Google APIs](https://developers.googl
...
+that enable the feature.
+
+#### Configuration
+
+- `provider` `(string: <required>)` - Name of the provider. Must be set to "gsuite".
+- `gsuite_service_account` `(string: <required>)` - Path to the Google service account key file obtained from setup.
+- `gsuite_admin_impersonate` `(string: <required>)` - Email address of a G Suite admin to impersonate.
+- `fetch_groups` `(bool: false)` - If set to true, groups will be fetched from G Suite.
+- `fetch_user_info` `(bool: false)` - If set to true, user info will be fetched from G Suite using the configured [user_custom_schemas](#user_custom_schemas).
+- `groups_recurse_max_depth` `(int: <optional>` - Group membership recursion max depth. Defaults to 0, which means don't recurse.
```
Comment (kalafut) [inline website/pages/docs/auth/jwt_oidc_providers.mdx:169]:
```suggestion
- `groups_recurse_max_depth` `(int: <optional>)` - Group membership recursion max depth. Defaults to 0, which means don't recurse.
```

File: website/pages/docs/auth/jwt_oidc_providers.mdx [hunk 3]
```diff
@@ -128,6 +128,79 @@ Main reference: [Using OAuth 2.0 to Access Google APIs](https://developers.googl
...
+- `user_custom_schemas` `(string: <optional>` - Comma-separated list of G Suite [custom schemas](https://developers.google.com/admin-sdk/directory/v1/guides/manage-schemas).
```
Comment (kalafut) [inline website/pages/docs/auth/jwt_oidc_providers.mdx:170]:
```suggestion
- `user_custom_schemas` `(string: <optional>)` - Comma-separated list of G Suite [custom schemas](https://developers.google.com/admin-sdk/directory/v1/guides/manage-schemas).
```

**Comment by kalafut** [comment]
LGTM. Let's hold the merge until the timing is finalize. Thanks!

File: website/pages/docs/auth/jwt_oidc_providers.mdx [hunk 4]
```diff
@@ -128,6 +128,79 @@ Main reference: [Using OAuth 2.0 to Access Google APIs](https://developers.googl
...
+[Groups](https://developers.google.com/admin-sdk/directory/v1/guides/manage-groups) APIs.
+
+In **step 5** within the section titled
+[Delegate domain-wide authority to your service account](https://developers.google.com/admin-sdk/directory/v1/guides/delegation#delegate_domain-wide_authority_to_your_service_account),
+the only OAuth scopes that should be granted are:
+- `https://www.googleapis.com/auth/admin.directory.group.readonly`
+- `https://www.googleapis.com/auth/admin.directory.user.readonly`
+
+This is an **important security step** in order to give the service account the least set of privileges
+that enable the feature.
```
Comment (tvoran) [inline website/pages/docs/auth/jwt_oidc_providers.mdx:160]: I wonder if it'd be worth calling this out a little more by using the `~>` syntax, like we've done here: 

https://github.com/hashicorp/vault/blob/4c44ccd52ab5bb7bb43d4a16235a18bf7e9f9634/website/pages/docs/platform/k8s/helm/configuration.mdx#L124-L125

Just a thought, I think it's also good the way it is.

Comment (austingebauer) [reply website/pages/docs/auth/jwt_oidc_providers.mdx]: Good idea. I like that better. Done in d6b6f6909e41e1862b8718b3431c4e3bbb3df722.

**Pull request merged**